src/Layer/C3DTilesLayer.js

```diff
@@ -142,48 +409,152 @@ class C3DTilesLayer extends GeometryLayer {
-                // get batch id of the face
-                batchID = interAttributes._BATCHID.array[vertex];
-                batchTable = this.findBatchTable(intersects[i].object);
+    getC3DTileFeatureFromIntersectsArray(intersects) {
+        // find closest intersect with an attributes _BATCHID + face != undefined
+        let closestIntersect = null;
+        for (let index = 0; index < intersects.length; index++) {
+            const i = intersects[index];
+            if (i.object.geometry &&
+                i.object.geometry.attributes._BATCHID &&
+                i.face && // need face to get batch id
+                i.layer == this // just to be sure that the right layer intersected
+            ) {
+                closestIntersect = i;
                 break;
             }
         }
 
-        if (batchID === -1) {
-            return;
+        if (!closestIntersect) {
+            return null;
         }
 
-        resultInfo.batchID = batchID;
-        // get information from batch table (including from its extension)
-        Object.assign(resultInfo, batchTable.getInfoById(batchID));
+        // function find the tile id from an object
+        function findTileID(object) {
+            let currentObject = object;
+            let result = currentObject.tileId;
+            while (isNaN(result) && currentObject.parent) {
+                currentObject = currentObject.parent;
+                result = currentObject.tileId;
+            }
+
+            return result;
+        }
+
+        const tileId = findTileID(closestIntersect.object);
+        // face is a Face3 object of THREE which is a
+        // triangular face. face.a is its first vertex
+        const vertex = closestIntersect.face.a;
+        const batchID = closestIntersect.object.geometry.attributes._BATCHID.array[vertex];
+
+        return this.tilesC3DTileFeatures.get(tileId).get(batchID);
+    }
+
+    /**
+     * Call by 3DtilesProcessing {@link 3dTilesProcessing} which handle load and unload of 3DTiles
+     * @param {THREE.Object3D} tileContent - tile as THREE.Object3D
+     */
+    onTileContentLoaded(tileContent) {
+        this.initC3DTileFeatures(tileContent);
+
+        // notify observer
+        this.dispatchEvent({ type: C3DTilesLayer.EVENT_TILE_CONTENT_LOADED, tileContent });
+    }
+
+    /**
+     * Initialize C3DTileFeatures from tileContent
+     *
+     * @param {THREE.Object3D} tileContent - tile as THREE.Object3D
+     */
+    initC3DTileFeatures(tileContent) {
+        tileContent.traverse((child) => {
+            if (child.geometry && child.geometry.attributes._BATCHID) {
+                this.tilesC3DTileFeatures.set(tileContent.tileId, new Map());// initialize
+                const batchTable = this.findBatchTable(child);
+                if (!batchTable) {
+                    throw new Error('no batchTable');
+                }
+
+                const geoAttributes = child.geometry.attributes;
```

Comment:
`geomAttributs` to avoid confusion with `geo` as `geography` ? ^^

Reply:
done